playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerView.java

```diff
@@ -437,5 +448,40 @@ public void setSurfaceAspectRatioResizeMode(PKAspectRatioResizeMode resizeMode)
         }
         return exoPlayerAspectRatioResizeMode;
     }
+
+    /**
+     * Creates new cue configuration if `isIgnoreCueSettings` is set to true by application
+     * Checks if the application wants to ignore the in-stream CueSettings otherwise goes with existing Cue configuration
+     *
+     * @param cueList cue list coming in stream
+     * @param subtitleViewPosition subtitle view position configuration set by application
+     * @return List of modified Cues
+     */
+    public List<Cue> getModifiedSubtitlePosition(List<Cue> cueList, PKSubtitlePosition subtitleViewPosition) {
+        if (!cueList.isEmpty()) {
+            for (int cuePosition = 0; cuePosition < cueList.size(); cuePosition++) {
+                Cue cue = cueList.get(cuePosition);
+                if ((cue.line !=  Cue.DIMEN_UNSET || cue.position != Cue.DIMEN_UNSET)
+                        && !subtitleViewPosition.isOverrideInlineCueConfig()) {
+                    continue;
+                }
+                CharSequence text = cue.text;
+                if (text != null) {
+                    Cue newCue = new Cue(text,
+                            subtitleViewPosition.getSubtitleHorizontalPosition(),
+                            subtitleViewPosition.getVerticalPositionPercentage(), // line and line type are dependent
+                            subtitleViewPosition.getLineType(),
+                            cue.lineAnchor,
+                            cue.position,
+                            cue.positionAnchor,
+                            subtitleViewPosition.getHorizontalPositionPercentage());
+                    cueList.clear();
```

Comment:
@GouravSna 
you should only clear the current Cue Index and not the whole list

Reply:
Done